src/server/scripts/Kalimdor/zone_orgrimmar.cpp

```diff
@@ -263,8 +264,563 @@ class npc_thrall_warchief : public CreatureScript
+                        break;
+                    case EVENT_HERALD_SCENE13:
+                        if (Creature* jaina = ObjectAccessor::GetCreature(*me, jainaGUID))
+                            jaina->AI()->Talk(SAY_JAINA_3);
+                        events.ScheduleEvent(EVENT_HERALD_SCENE14, 10000);
+                        break;
+                    case EVENT_HERALD_SCENE14:
+                        if (Creature* jaina = ObjectAccessor::GetCreature(*me, jainaGUID))
+                        {
+                            jaina->AI()->Talk(SAY_JAINA_4);
+                            jaina->GetMotionMaster()->MoveTargetedHome();
+                            jaina->DespawnOrUnsummon(5000);
+                        }
+                        events.ScheduleEvent(EVENT_HERALD_SCENE15, 7000);
+                        break;
+                    case EVENT_HERALD_SCENE15:
+                        me->GetMotionMaster()->MoveTargetedHome();
+                        Talk(SAY_THRALL_7);
+
+                        if (Creature* sylvanas = ObjectAccessor::GetCreature(*me, sylvanasGUID))
+                            sylvanas->GetMotionMaster()->MoveTargetedHome();
+
+                        if (Creature* portal = ObjectAccessor::GetCreature(*me, stormwindPortalGUID))
+                            portal->DespawnOrUnsummon();
+
+                        if (GameObject* leftPortal = ObjectAccessor::GetGameObject(*me, ucPortalGUIDs[0]))
+                            leftPortal->RemoveFromWorld();
+
+                        if (GameObject* rightPortal = ObjectAccessor::GetGameObject(*me, ucPortalGUIDs[1]))
+                            rightPortal->RemoveFlag(GAMEOBJECT_FLAGS, GO_FLAG_NOT_SELECTABLE);
+
+                        events.ScheduleEvent(EVENT_HERALD_RESET, 60000);
+                        break;
+                    case EVENT_HERALD_RESET:
+                        Initialize();
+                        break;
+                    default:
+                        break;
+                }
+            }
+        }
+
+    private:
+        EventMap events;
+        ObjectGuid guardsGUIDs[4];
+        ObjectGuid jainaGUID;
+        ObjectGuid sylvanasGUID;
+        ObjectGuid ucPortalGUIDs[2];
+        ObjectGuid stormwindPortalGUID;
+        bool spawnedGuards;
+        bool sceneInProgress;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_thrall_herald_of_warAI(creature);
+    }
+};
+
+class areatrigger_orgrimmar_herald_of_war : public AreaTriggerScript
+{
+public:
+    areatrigger_orgrimmar_herald_of_war() : AreaTriggerScript("areatrigger_orgrimmar_herald_of_war") { }
+
+    bool OnTrigger(Player* player, AreaTriggerEntry const* /*trigger*/) override
+    {
+        if (player->IsAlive() && player->GetQuestStatus(QUEST_HERALD_OF_WAR) == QUEST_STATUS_COMPLETE && player->GetQuestStatus(QUEST_HERALD_OF_WAR) != QUEST_STATUS_REWARDED)
```

Comment:
I see player->GetQuestStatus(QUEST_HERALD_OF_WAR) is checked against == QUEST_STATUS_COMPLETE and != QUEST_STATUS_REWARDED , is there a typo or is one of the checks not needed ?